hil/client/base.py

```diff
@@ -76,12 +76,12 @@ def check_response(self, response):
 
 def _find_reserved(string, slashes_ok=False):
     """Returns a list of illegal characters in a string"""
-    if slashes_ok:
-        p = r"[^A-Za-z0-9 /$_.+!*'(),-]+"
-    else:
-        p = r"[^A-Za-z0-9 $_.+!*'(),-]+"
-    # return all unique chars that exist neither in p nor string
-    return list(set(x for l in re.findall(p, string) for x in l))
+    p = r"[^A-Za-z0-9 /$_.+!*'(),-]"
```

Comment:
This should probably *not* include the slash I think? Otherwise the slashes will end up in the result regardless of the value of `slashes_ok`. Would you also add a test that catches this?

Reply:
I thought that result shouldn't check slashes by default so I put one in my "allowed characters". Then, if slashes aren't allowed but there is one in the string, I need to add a slash to my list of unique illegal characters, result. My goal is for result to have all the characters that are in string but not in p.

My tests all failed when I did remove the slash from p, but when I put it back everything worked as it should.  Tomorrow I'll double check that and write a test to check for a proper response.

Reply:
Ah, no you're totally right; I had things backwards in my head; forgot that the regex was being inverted, so would match allowed characters rather than banned characters.